config/routes.rb

```diff
@@ -149,6 +149,9 @@
   %w[terms syarat ketentuan].each { |r| get "/#{r}", to: 'home#terms' }
   %w[contact kontak hubungi].each { |r| get "/#{r}", to: 'home#contact' }
 
+  get '/unsubscribe/:token', to: 'user_notifications#unsubscribe', as: :unsubscribe
```

Comment:
Can you look into the patterns above and use them? i.e. `resources`, `collection`, etc.

Reply:
i want to make sure this route, is this right?
```
resources :users do
    resources :user_notifications, path: '/user-notifications',
                                   only: %i[index create] do
      collection do
        delete 'delete', to: 'user_notifications#delete'
        get '/unsubscribe_from_all_notifications/:token', to: 'user_notifications#unsubscribe_from_all_notifications',
                                                          as: :unsubscribe_from_all_notifications
        get '/unsubscribe_from_one_notification/:token/', to: 'user_notifications#unsubscribe_from_one_notification',
                                                          as: :unsubscribe_from_one_notification
      end
    end
  end
```

Reply:
if true, then unsubscribe links will need user_id which i don't know how to present it because only tokens that exist

Reply:
You have the user notification right? Can't you just do `user_notification.user_id`?

Reply:
i didn't understand, can u help me explain it with more detail, please? @donjar

Reply:
You need the `user_id` right? When you construct the route, you ought to have the `user_notification` object because you have the token. So I'm saying that you can get the `user_id` from the `user_notification` object.